astroplan/constraints.py

```diff
@@ -0,0 +1,614 @@
+        solar_altitude = sun_altaz['altitude']
+        mask = solar_altitude < self.max_solar_altitude
+        return mask
+
+class SunSeparationConstraint(Constraint):
+    """
+    Constrain the distance between the Sun and some targets.
+    """
+    def __init__(self, min=None, max=None):
+        """
+        Parameters
+        ----------
+        min : `~astropy.units.Quantity` or `None` (optional)
+            Minimum acceptable separation between Sun and target. `None`
+            indicates no limit.
+        max : `~astropy.units.Quantity` or `None` (optional)
+            Minimum acceptable separation between Sun and target. `None`
+            indicates no limit.
+        """
+        self.min = min
+        self.max = max
+
+    def compute_constraint(self, times, observer, targets):
+        sun = get_sun(times)
+        targets = [target.coord if hasattr(target, 'coord') else target
+                   for target in targets]
+        solar_separation = Angle([sun.separation(target) for target in targets])
+        if self.min is None and self.max is not None:
+            mask = self.max > solar_separation
+        elif self.max is None and self.min is not None:
+            mask = self.min < solar_separation
+        elif self.min is not None and self.max is not None:
+            mask = ((self.min < solar_separation) &
+                    (solar_separation < self.max))
+        else:
+            raise ValueError("No max and/or min specified in "
+                             "SunSeparationConstraint.")
+        return mask
+
+class MoonSeparationConstraint(Constraint):
+    """
+    Constrain the distance between the Earth's moon and some targets.
+    """
+    def __init__(self, min=None, max=None):
+        """
+        Parameters
+        ----------
+        min : `~astropy.units.Quantity` or `None` (optional)
+            Minimum acceptable separation between moon and target. `None`
+            indicates no limit.
+        max : `~astropy.units.Quantity` or `None` (optional)
+            Minimum acceptable separation between moon and target. `None`
+            indicates no limit.
+        """
+        self.min = min
+        self.max = max
+
+    def compute_constraint(self, times, observer, targets):
+        moon = get_moon(times, observer.location, observer.pressure)
+        targets = [target.coord if hasattr(target, 'coord') else target
+                   for target in targets]
+
+        moon_separation = Angle([angular_separation(moon.spherical.lon,
+                                                    moon.spherical.lat,
+                                                    target.spherical.lon,
+                                                    target.spherical.lat)
+                                 for target in targets])
+        # The line below should have worked, but needs a workaround.
+        # TODO: once bug has been fixed, replace workaround with simpler version.
```

Comment:
Please either fix this TODO in this PR or say which bug you're referring to here.